core/src/client/mod.rs

```diff
@@ -343,29 +371,52 @@ where
 	Notif: DeserializeOwned,
 {
 	/// Returns the next notification from the stream.
-	/// This may return `None` if the subscription has been terminated,
-	/// which may happen if the channel becomes full or is dropped.
+	/// 
+	/// The return values can be:
+	/// - `Some(Ok(val)`: The message was succesfully received and decoded.
+	/// - `Some(Err(SubscriptionError::Lagged(n)))`: The subscription buffer was full and n messages were lost.
+	/// - `Some(Err(SubscriptionError::Parse: the value failed to be decode as `Notif`
+	/// - `None`: The connection was closed.
```

Comment:
So, `None` is returned when: (a) connection is dropped; and (b) buffer is drained?

Reply:
No, it should only return None when the stream is "closed" 

https://docs.rs/tokio-stream/latest/src/tokio_stream/wrappers/broadcast.rs.html#54-67